core/src/main/scala/com/microsoft/azure/synapse/ml/codegen/DotnetWrappable.scala

```diff
@@ -0,0 +1,322 @@
+        |/// </summary>
+        |/// <returns>an <see cref=\"JavaMLReader&lt;$dotnetClassName&gt;\"/> instance for this ML instance.</returns>
+        |public JavaMLReader<$dotnetClassName> Read() =>
+        |    new JavaMLReader<$dotnetClassName>((JvmObjectReference)Reference.Invoke("read"));
+        |""".stripMargin
+  }
+
+  protected def dotnetWrapAsTypeMethod: String = {
+    s"""|private static $dotnetClassName $dotnetClassWrapperName(object obj) =>
+        |    new $dotnetClassName((JvmObjectReference)obj);
+        |""".stripMargin
+  }
+
+  def dotnetAdditionalMethods: String = {
+    ""
+  }
+
+  //noinspection ScalaStyle
+  protected def dotnetParamSetter(p: Param[_]): String = {
+    val capName = p.name match {
+      case "xgboostDartMode" => "XGBoostDartMode"
+      case "parallelism" => dotnetClassName match {
+        case "VowpalWabbitContextualBandit" => "ParallelismForParamListFit"
+        case _ => p.name.capitalize
+      }
+      case _ => p.name.capitalize
+    }
+    val docString =
+      s"""|/// <summary>
+          |/// Sets ${p.name} value for <see cref=\"${p.name}\"/>
+          |/// </summary>
+          |/// <param name=\"${p.name}\">
+          |/// ${p.doc}
+          |/// </param>
+          |/// <returns> New $dotnetClassName object </returns>""".stripMargin
+    p match {
+      case sp: ServiceParam[_] =>
+        s"""|$docString
+            |public $dotnetClassName Set$capName(${getServiceParamInfo(sp).dotnetType} value) =>
+            |    $dotnetClassWrapperName(Reference.Invoke(\"set$capName\", (object)value));
+            |
+            |public $dotnetClassName Set${capName}Col(string value) =>
+            |    $dotnetClassWrapperName(Reference.Invoke(\"set${capName}Col\", value));
+            |""".stripMargin
+      // TODO: Fix UDF & UDPyF confusion; ParamSpaceParam, BallTreeParam, ConditionalBallTreeParam type
+      case wp: WrappableParam[_] =>
+        s"""|$docString
+            |${wp.dotnetSetter(dotnetClassName, capName, dotnetClassWrapperName)}
+            |""".stripMargin
+      case _ =>
+        s"""|$docString
+            |public $dotnetClassName Set$capName(${getParamInfo(p).dotnetType} value) =>
+            |    $dotnetClassWrapperName(Reference.Invoke(\"set$capName\", (object)value));
+            |""".stripMargin
+    }
+  }
+
+  protected def dotnetParamSetters: String =
+    thisStage.params.map(dotnetParamSetter).mkString("\n")
+
+  //noinspection ScalaStyle
+  protected def dotnetParamGetter(p: Param[_]): String = {
+    val capName = p.name.capitalize
+    val docString =
+      s"""|/// <summary>
+          |/// Gets ${p.name} value for <see cref=\"${p.name}\"/>
+          |/// </summary>
+          |/// <returns>
+          |/// ${p.name}: ${p.doc}
+          |/// </returns>""".stripMargin
+    p match {
+      case sp: ServiceParam[_] =>
+        val dType = getGeneralParamInfo(sp).dotnetType
+        dType match {
+          case "TimeSeriesPoint[]" |
```

Comment:
I moved this inside ServiceParam but we can't change the string matching to real type matching, because the params exist in cognitive package and we can't import them into core package.